tests/python/unittest/test_subgraph_op.py

```diff
@@ -146,11 +146,137 @@ def get_executor(sym, subgraph_backend=None, op_names=None, original_exec=None):
+        aux_array = [mx.nd.random.uniform(shape=shape) for shape in aux_shapes]
+        exe1 = sym.bind(ctx=mx.current_context(), args=arg_array, aux_states=aux_array, grad_req='null')
+        exe1.forward()
+
+        # partition before bind
+        check_call(_LIB.MXSetSubgraphPropertyOpNames(c_str(subgraph_backend), mx_uint(len(op_names)),
+                                                     c_str_array(op_names)))
+        part_sym = sym.optimize_for(subgraph_backend)
+        check_call(_LIB.MXRemoveSubgraphPropertyOpNames(c_str(subgraph_backend)))
+
+        exe2 = part_sym.bind(ctx=mx.current_context(), args=arg_array, aux_states=aux_array, grad_req='null')
+        exe2.forward()
+
+        # compare outputs
+        outputs1 = exe1.outputs
+        outputs2 = exe2.outputs
+        assert len(outputs1) == len(outputs2)
+        for i in range(len(outputs1)):
+            assert_almost_equal((outputs1[i] - outputs2[i]).abs().sum().asnumpy(), np.zeros(shape=(1,)))
+
+    def _check_subgraph_exe8(sym, subgraph_backend, op_names):
+        """Call optimize_for to infer shapes, types and dtypes followed by graph partitioning,
+        then bind and compare results of the partitioned sym and the original sym."""
+        # bind
+        arg_shapes, _, aux_shapes = sym.infer_shape()
+        arg_array = [mx.nd.random.uniform(shape=shape) for shape in arg_shapes]
+        aux_array = [mx.nd.random.uniform(shape=shape) for shape in aux_shapes]
+        exe1 = sym.bind(ctx=mx.current_context(), args=arg_array, aux_states=aux_array, grad_req='null')
+        exe1.forward()
+
+        # infer shape/type before partition before bind
+        check_call(_LIB.MXSetSubgraphPropertyOpNames(c_str(subgraph_backend), mx_uint(len(op_names)),
+                                                     c_str_array(op_names)))
+        part_sym = sym.optimize_for(subgraph_backend, arg_array)
+        check_call(_LIB.MXRemoveSubgraphPropertyOpNames(c_str(subgraph_backend)))
+
+        exe2 = part_sym.bind(ctx=mx.current_context(), args=arg_array, aux_states=aux_array, grad_req='null')
+        exe2.forward()
+
+        # compare outputs
+        outputs1 = exe1.outputs
+        outputs2 = exe2.outputs
+        assert len(outputs1) == len(outputs2)
+        for i in range(len(outputs1)):
+            assert_almost_equal((outputs1[i] - outputs2[i]).abs().sum().asnumpy(), np.zeros(shape=(1,)))
+
     def check_subgraph_exe(sym, subgraph_backend, op_names):
         _check_subgraph_exe1(sym, subgraph_backend, op_names)
         _check_subgraph_exe2(sym, subgraph_backend, op_names)
         _check_subgraph_exe3(sym, subgraph_backend, op_names)
         _check_subgraph_exe4(sym, subgraph_backend, op_names)
+        _check_subgraph_exe5(sym, subgraph_backend, op_names)
+        _check_subgraph_exe6(sym, subgraph_backend, op_names)
+        _check_subgraph_exe7(sym, subgraph_backend, op_names)
+        _check_subgraph_exe8(sym, subgraph_backend, op_names)
```

Comment:
should we separate those tests into a separate test file? since the previous tests are testing the 'subgraph building' but the new tests are testing partition api

Reply:
Its the same test. subgraph building is partitioning.